qiita_pet/util.py

```diff
@@ -116,3 +116,45 @@ def is_localhost(host):
     """
     localhost = ('localhost', '127.0.0.1')
     return host.startswith(localhost)
+
+
+def get_artifact_processing_status(artifact):
+    """Gets the processing status of the artifact
+
+    Parameters
+    ----------
+    artifact : qiita_db.artifact.Artifact
+        The artifact to get the processing status
+
+    Returns
+    -------
+    str, str
+        The processing status {'processing', 'failed', 'success',
+            'Not processed'}
+        A summary of the jobs attached to the artifact
+    """
+    preprocessing_status = 'Not processed'
+    preprocessing_status_msg = []
+    for job in artifact.jobs():
+        job_status = job.status
+        if job_status == 'error':
+            if preprocessing_status != 'success':
+                preprocessing_status = 'failed'
+            preprocessing_status_msg.append(
+                "<b>Job %s</b>: failed - %s"
+                % (job.id, job.log.msg))
+        elif job_status == 'success':
+            preprocessing_status = 'success'
```

Comment:
Yes - unfortunately, the current interface is designed in a way that only one job can be executed, so whenever we have a successful job we disallow launching any other job. This is just an interface limitation, since the underlying system supports way more. However, since we're not changing the interface in this PR, we need to emulate that behavior like that in here.